examples/ibm-power/main.tf

```diff
@@ -52,13 +66,27 @@ resource "ibm_pi_instance" "instance" {
   pi_sys_type          = var.sys_type
   pi_image_id          = data.ibm_pi_image.data_source_image.id
   pi_key_pair_name     = data.ibm_pi_key.data_source_key.id
-  pi_network { network_id = data.ibm_pi_public_network.data_source_network.id }
+  pi_network {
+    network_id = data.ibm_pi_network.data_source_private_network.id
+  }
   pi_volume_ids = [data.ibm_pi_volume.data_source_volume.id]
 }
-
 data "ibm_pi_instance" "data_source_instance" {
   depends_on = [ibm_pi_instance.instance]
 
-  pi_cloud_instance_id = var.cloud_instance_id
+  pi_cloud_instance_id = ibm_resource_instance.location.guid
   pi_instance_name     = var.instance_name
-}
\ No newline at end of file
+}
+
+# Create an ssh key
+resource "ibm_pi_key" "key" {
+  pi_cloud_instance_id = ibm_resource_instance.location.guid
+  pi_key_name          = var.ssh_key_name
+  pi_ssh_key           = var.ssh_key_rsa
+}
+data "ibm_pi_key" "data_source_key" {
+  depends_on = [ibm_pi_key.key]
+
+  pi_cloud_instance_id = ibm_resource_instance.location.guid
+  pi_key_name          = var.ssh_key_name
+}
```

Comment:
This should go above `resource "ibm_pi_instance" "instance"{...}`

Reply:
Shifted position.